File: Top 50 Coding Interview Questions/50. Write a code to find Fibonacci Series using Recursion/FibonacciSeriesusingRecursion.py
# Function to calculate the nth Fibonacci number
def Fibo(n):
    """
    Calculates the nth Fibonacci number using recursion.

    Args:
    n (int): The position in the Fibonacci sequence (0-indexed).

    Returns:
    int: The nth Fibonacci number.
    """
    if n < 0:
        print("Incorrect input! Please enter a non-negative integer.")
        return None  # Return None for invalid input
    elif n == 0:
        return 0  # 1st Fibonacci number is 0
    elif n == 1:
        return 1  # 2nd Fibonacci number is 1
    else:
        # Recursive case
        return Fibo(n - 1) + Fibo(n - 2)


# Main Program (without if __name__ == "__main__":)
n = 9
print(f"The {n}th Fibonacci number is: {Fibo(n)}")

File: Top 50 Coding Interview Questions/50. Write a code to find Fibonacci Series using Recursion/FibonacciSeriesusingRecursion.cpp
#include <iostream>
using namespace std;

// Recursive function to calculate the nth Fibonacci number
int fibonacci(int n) {
    if (n <= 1) {
        return n; // Base cases: Fib(0) = 0, Fib(1) = 1
    }
    return fibonacci(n - 1) + fibonacci(n - 2); // Recursive case
}

int main() {
    // Take user input for the position in the Fibonacci sequence
    int n;
    cout << "Enter the position (n) for Fibonacci series: ";
    cin >> n;

    // Validate input
    if (n < 0) {
        cout << "Invalid input! Please enter a non-negative integer." << endl;
        return 1; // Exit with error code
    }

    // Calculate and display the nth Fibonacci number
    cout << "The " << n << "th Fibonacci number is: " << fibonacci(n) << endl;

    return 0;
}
